tests/components/hyperion/test_light.py

```diff
@@ -27,98 +27,84 @@ def create_mock_client():
+    return client.set_callbacks.call_args[0][0][key](*args, **kwargs)
 
 
-def create_client_and_entity():
-    """Create a mock Hyperion client and entity."""
-    client = create_mock_client()
-    return client, create_hyperion_entity(client)
+async def setup_entity(hass, client=None):
+    """Add a test Hyperion entity to hass."""
+    client = client or create_mock_client()
+    with patch("hyperion.client.HyperionClient", return_value=client):
+        assert await async_setup_component(
+            hass,
+            DOMAIN,
+            {
+                DOMAIN: {
+                    "platform": "hyperion",
+                    "name": TEST_NAME,
+                    "host": TEST_HOST,
+                    "port": const.DEFAULT_PORT,
+                    "priority": TEST_PRIORITY,
+                }
+            },
+        )
+    await hass.async_block_till_done()
 
 
 async def test_setup_platform(hass):
-    """Test the platform setup."""
-    add_entities = Mock()
+    """Test setting up the platform."""
     client = create_mock_client()
-
-    with patch("hyperion.client.HyperionClient", return_value=client):
-        await hyperion_light.async_setup_platform(hass, TEST_CONFIG, add_entities)
-        await hass.async_block_till_done()
+    await setup_entity(hass, client=client)
 
     # Make sure the background task is initiated.
     assert client.start_background_task.called
-
-    # Make sure the entity is added.
-    assert add_entities.called
+    assert hass.states.get(TEST_ENTITY_ID) is not None
 
 
 async def test_setup_platform_not_ready(hass):
     """Test the platform not being ready."""
-    add_entities = Mock()
     client = create_mock_client()
     client.async_connect = CoroutineMock(return_value=False)
 
-    with pytest.raises(PlatformNotReady):
-        with patch("hyperion.client.HyperionClient", return_value=client):
-            await hyperion_light.async_setup_platform(hass, TEST_CONFIG, add_entities)
-            await hass.async_block_till_done()
+    await setup_entity(hass, client=client)
 
     # Make sure the background task is initiated.
     assert not client.start_background_task.called
-
-    # Make sure the entity is added.
-    assert not add_entities.called
+    assert hass.states.get(TEST_ENTITY_ID) is None
 
 
 async def test_light_basic_properies(hass):
     """Test the basic properties."""
-    client, entity = create_client_and_entity()
-    assert entity
-    assert not entity.should_poll
-    assert entity.name == TEST_NAME
-    assert entity.brightness == 255
-    assert entity.hs_color == (0.0, 0.0)
-
-    client.is_on.return_value = False
-    assert not entity.is_on
-
-    client.is_on.return_value = True
-    assert entity.is_on
+    client = create_mock_client()
+    await setup_entity(hass, client=client)
 
-    assert entity.icon == hyperion_light.ICON_LIGHTBULB
-    assert entity.effect == hyperion_light.KEY_EFFECT_SOLID
+    entity = hass.states.get(TEST_ENTITY_ID)
```

Comment:
This is a state instance. Please rename it to avoid confusion. Eg:
```suggestion
    entity_state = hass.states.get(TEST_ENTITY_ID)
```